networking/eth2/src/test/java/tech/pegasys/artemis/networking/eth2/gossip/BlockValidatorTest.java

```diff
@@ -0,0 +1,69 @@
+package tech.pegasys.artemis.networking.eth2.gossip;
+
+import com.google.common.eventbus.EventBus;
+import com.google.common.primitives.UnsignedLong;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import tech.pegasys.artemis.bls.BLSKeyGenerator;
+import tech.pegasys.artemis.bls.BLSKeyPair;
+import tech.pegasys.artemis.core.ChainBuilder;
+import tech.pegasys.artemis.core.StateTransition;
+import tech.pegasys.artemis.core.StateTransitionException;
+import tech.pegasys.artemis.datastructures.blocks.SignedBeaconBlock;
+import tech.pegasys.artemis.networking.eth2.gossip.topics.validation.BlockValidationResult;
+import tech.pegasys.artemis.networking.eth2.gossip.topics.validation.BlockValidator;
+import tech.pegasys.artemis.statetransition.BeaconChainUtil;
+import tech.pegasys.artemis.storage.client.MemoryOnlyRecentChainData;
+import tech.pegasys.artemis.storage.client.RecentChainData;
+
+import java.util.List;
+
+import static com.google.common.primitives.UnsignedLong.ONE;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static tech.pegasys.artemis.datastructures.util.BeaconStateUtil.compute_start_slot_at_epoch;
+
+public class BlockValidatorTest {
```

Comment:
Looks like a few test cases are missing for invalid signatures, missing prestate, duplicate block

Reply:
Yeah, I put it up as a draft PR to get feedback. Gotta figure out how to make PR's actually draft.